src/sdam/topology.ts

```diff
@@ -473,17 +473,13 @@ export class Topology extends TypedEventEmitter<TopologyEvents> {
 
         const skipPingOnConnect = this.s.options[Symbol.for('@@mdb.skipPingOnConnect')] === true;
         if (!skipPingOnConnect && server && this.s.credentials) {
-          server.command(ns('admin.$cmd'), { ping: 1 }, {}, err => {
-            if (err) {
-              return exitWithError(err);
-            }
-
+          server.command(ns('admin.$cmd'), { ping: 1 }, {}).then(() => {
```

Comment:
seems like it would be minimal effort to just make `Topology.connect()` async.  Thoughts?

Reply:
https://jira.mongodb.org/browse/NODE-5914